transport/rif_comms/node.py

```diff
@@ -176,23 +177,21 @@ def stop(self):
             if message_queue.greenlet:
                 message_queue.notify()  # if we need to send something, this is the time
 
-        if self.greenlet:
-            self.greenlet.kill()
-            self.greenlet.get()
-
-        # wait for our own greenlets, no need to get on them, exceptions should be raised in _run()
-        wait([self.greenlet] + [r.greenlet for r in self._address_to_message_queue.values()])
-
-        self._comms_client.disconnect()
+        # wait for message queues, self.greenlet just dispatch the handle message, so there is no point of wait for it.
+        wait([r.greenlet for r in self._address_to_message_queue.values()])
 
         self.log.debug("RIF Comms Node stop", config=self._config)
         try:
             del self._log
         except AttributeError:
             # During shutdown the log attribute may have already been collected
             pass
-        # parent may want to call get() after stop(), to ensure _run errors are re-raised
-        # we don't call it here to avoid deadlock when self crashes and calls stop() on finally
+
+        # end grpc communication
+        self._comms_client.disconnect()
```

Comment:
it's possible we'll have to change this if we go by my previous suggestion regarding the client's code

Reply:
yes indeed, on it